src/FlightDisplay/GuidedAltitudeSlider.qml

```diff
@@ -62,20 +70,22 @@ Rectangle {
             anchors.right:          parent.right
             wrapMode:               Text.WordWrap
             horizontalAlignment:    Text.AlignHCenter
-            text:                   qsTr("New Alt(rel)")
+            text:                   _useAMSL ? qsTr("New Alt(AMSL)") : qsTr("New Alt(rel)")
+            color:                  altField.color
         }
 
         QGCLabel {
             id:                         altField
             anchors.horizontalCenter:   parent.horizontalCenter
             text:                       newAltitudeAppUnits + " " + QGroundControl.unitsConversion.appSettingsHorizontalDistanceUnitsString
+            color:                      !isNaN(_distanceToGround) && (-altLossGain > _distanceToGround) ? qgcPal.warningText : qgcPal.text
 
-            property real   altGainRange:           Math.max(_sliderMaxAlt - _vehicleAltitude, 0)
-            property real   altLossRange:           Math.max(_vehicleAltitude - _sliderMinAlt, 0)
             property real   altExp:                 Math.pow(altSlider.value, 3)
             property real   altLossGain:            altExp * (altSlider.value > 0 ? altGainRange : altLossRange)
             property real   newAltitudeMeters:      _vehicleAltitude + altLossGain
-            property string newAltitudeAppUnits:    QGroundControl.unitsConversion.metersToAppSettingsHorizontalDistanceUnits(newAltitudeMeters).toFixed(1)
+            property real   newAltitudeAMSLMeters:  _vehicleAltitudeAMSL + altLossGain
+            property string newAltitudeAppUnits:    QGroundControl.unitsConversion.metersToAppSettingsHorizontalDistanceUnits(
```

Comment:
I would guess VerticalDistanceUnits, but not important

Reply:
Same issue as in step buttons, for consistency it is better to use the wrong type.